Update documentation for taxes (#4405)
mirumee/saleor: docs/guides/taxes.rst

Taxes
=====

> Saleor gives a possibility to configure taxes. It supports external tax provides: ``Avarala``, ``Vatlayer``. Currently only one provider can be enabled at the same time.

Reviewer: ```suggestion
Saleor provides the possibility to configure taxes. It supports the following external tax providers: ``Avarala``, ``Vatlayer``. Currently only one provider can be enabled at the same time.
```

.. _taxes_configuration:

Taxes
=====

> Saleor provides the possibility to configure taxes. It supports the following external tax providers: ``Avarala``, ``Vatlayer``. Currently only one provider can be enabled at the same time.

Reviewer 2: ```suggestion
Saleor provides the possibility to configure taxes. It supports the following external tax providers: ``Avalara``, ``Vatlayer``. Currently, only one provider can be enabled at the same time.
```

#. Show gross prices to customers in the storefront

   If selected, prices displayed for customers in storefront will be gross (only ``Vatlayer``). Taxes will be properly calculated at checkout. Changing this setting has no effect on displaying orders placed in the past.

#. Charge taxes on shipping rates

   If selected, standard tax rate will be charged on shipping price.

Vatlayer
--------

> Configuration can be done in dashboard ``Taxes`` tab.

Reviewer 2: I would change the order of those three paragraphs. First "Taxes are charged..." then "Vatleyer handles taxes..." and lastly the part about configuration.

> Configuration can be done in dashboard ``Taxes`` tab.

Reviewer 2: ```suggestion
You can configure taxes calculations in the "Taxes" section in the Dashboard.
```

    By default, Saleor is making requests to the VatLayer API through HTTP (insecure),
    if you are using a paid VatLayer subscription, you may want to set the settings ``VATLAYER_USE_HTTPS`` to ``True``.

Avalara
-------

> All configuration and account setup should be done on the Avalara's admin panel. Saleor uses standard Avalata's  API to fetch tax codes, calculate taxes during the checkout process and transfer data directly to the Avalara's system.

Reviewer: ```suggestion
All configuration and account setup should be done on the Avalara's admin panel. Saleor uses the standard API of Avalata's to fetch tax codes, to calculate taxes during the checkout process and to transfer data directly to the Avalara's system.
```

Configuration
^^^^^^^^^^^^^

> Only basic data needs to be provided on Saleor side. You need to fill up variables:

Reviewer: ```suggestion
Only basic data needs to be provided on Saleor's side. You need to fill up the following variables:
```

> ``AVATAX_USERNAME_OR_ACCOUNT``, ``AVATAX_PASSWORD_OR_LICENSE``, ``AVATAX_USE_SANDBOX``, ``AVATAX_COMPANY_NAME``, ``AVATAX_AUTOCOMMIT``, (:ref:`see description of variables.<tax_environment_variables>`)

Reviewer: ```suggestion
``AVATAX_USERNAME_OR_ACCOUNT``, ``AVATAX_PASSWORD_OR_LICENSE``, ``AVATAX_USE_SANDBOX``, ``AVATAX_COMPANY_NAME``, ``AVATAX_AUTOCOMMIT``, (:ref:`refer to the variables descriptions. <tax_environment_variables>`)
```

Avalara requires a company address to properly calculate taxes.

> The address must be assigned in the dashboard's general section.

Reviewer: ```suggestion
The address must be assigned in the general section of Avalara's dashboard.
```

Author: It is in our dashboard not in Avalara's admin panel.

.. warning::

>   Avalara will not work without company address provided in the dashboard.

Reviewer: ```suggestion
  Avalara will not work without the company address provided in the dashboard.
```

Tax codes
^^^^^^^^^

> Avalara has own system for classification of product type and expected taxes. On Saleor side, store owner needs to assign Avalara's tax code to the given product type or product itself. If you skip this step, Saleor will use common code for all products (which is not recommended).

Reviewer: ```suggestion
Avalara has its own system for classifing the product types and expected taxes. On Saleor's side, the store owner must assign Avalara's tax code to the given product types or product itself. If you skip this step, Saleor will use the common code for all products (which is not recommended).
```

> Assigning a tax code can be done from dashboard in product type or product section. Both of these sections have a field for assigning tax code for enabled tax gateway.

Reviewer: ```suggestion
Assigning a tax code can be done from the dashboard in the product type or the product section. Both of these sections have a field for assigning a tax code for enabled tax gateway.
```

    By default, Saleor is making requests to the VatLayer API through HTTP (insecure),
    if you are using a paid VatLayer subscription, you may want to set the settings ``VATLAYER_USE_HTTPS`` to ``True``.

Avalara
-------

All configuration and account setup should be done on the Avalara's admin panel. Saleor uses the standard API of
Avalata's to fetch tax codes, to calculate taxes during the checkout process and to transfer data directly to the
Avalara's system.

Configuration
^^^^^^^^^^^^^

Only basic data needs to be provided on Saleor's side. You need to fill up the following variables:
``AVATAX_USERNAME_OR_ACCOUNT``, ``AVATAX_PASSWORD_OR_LICENSE``, ``AVATAX_USE_SANDBOX``, ``AVATAX_COMPANY_NAME``, ``AVATAX_AUTOCOMMIT`` (:ref:`refer to the variables descriptions.<tax_environment_variables>`)

Avalara supports two ways of authentications:
    1. Through login information (``username`` + ``password``)
    2. Through the account information (``account_id`` + ``license``)

Use ``AVATAX_USERNAME_OR_ACCOUNT`` and ``AVATAX_PASSWORD_OR_LICENSE`` to pass one of these authentication pairs

Avalara requires a company address to properly calculate taxes.

> The address must be assigned in general section of Saleor's dashboard.

Reviewer 2: ```suggestion
Your company's address can be assigned in the dashboard in "Configure" > "Site Settings" > "Company Information" section.
```

> .. warning::

Reviewer 2: We've just said in the paragraph above that the company address is required, so this seems unnecessary.
